spectral_cube/spectral_cube.py

```diff
@@ -1820,6 +1820,10 @@ def subcube(self, xlo='min', xhi='max', ylo='min', yhi='max', zlo='min',
         """
         Extract a sub-cube spatially and spectrally.
 
+        When spatial WCS dimensions are given, the spatial coordinates of the 'lo'
+        and 'hi' corners are solved together. This minimizes WCS variations due to
+        the sky curvature when slicing from a large (>1 deg) image.
```

Comment:
Naive question, but if xlo/xhi/ylo/yhi are all world coordinates, why not simply call world2pix for each of the four corners?

Reply:
Yeah, that's a lot simpler! I've added that in and kept the minimum distance solving only for the case of mixed pixel/WCS inputs.

Reply:
And we can revert back to a 1D WCS coord slice when the input for a corner is mixed (1 pixel, 1 WCS). So the full spatial coordinate comparison I had isn't needed anymore